hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1894,6 +1897,26 @@ public void updatePeerList(List<String> newPeers) {
         }
       }
     }
+    RaftPeer leader = null;
+    try {
+      leader = omRatisServer.getLeader();
+    } catch (IOException ex) {
+      LOG.error("IOException while getting the " +
+          "Ratis server leader.", ex);
+    }
+    if (Objects.nonNull(leader)) {
+      String leaderId = leader.getId().toString();
+
+      // If leaderId is empty, then leader is undefined
+      // and current OM is neither leader nor follower.
+      // OMHAMetrics shouldn't be registered in that case.
+      if (!Strings.isNullOrEmpty(leaderId)) {
```

Comment:
@neils-dev @xBis7

I had to update this part of the code after merging the PR due to a conflicting change that had been just merged (HDDS-6743, `getLeader()` no longer throws `IOException`).

Upon closer inspection, I think these `if-else` blocks have a problem.  Please correct me if I'm wrong, but if leader is undefined then `leader` will be `null`, so the unregistration will not happen.  (This seems to have been the case even before HDDS-6743.)  If we have any leader information, its `id` cannot be `null`.

I think it should be:

```java
if (Objects.nonNull(leader)) {
  // init
} else {
  // unregister
}
```

Reply:
@adoroszlai You are right. This was missed because the code was 

```java
String leaderId = "";
try{
  leader = 
} catch() {

}

if (Objects.nonNull(leader)) {

}

// leaderId could be deliberately left empty down here due to failure to get the leader
// after refactoring `String leaderId = "";` was removed.
```

> If we have any leader information, its id cannot be null.

I didn't know that.

How can we handle this now since the code has been merged?

Reply:
Thanks @xBis7 for checking.

> > If we have any leader information, its id cannot be null.
> 
> I didn't know that.

https://github.com/apache/ratis/blob/27f5a59cc697ce51f1c5ab6d7b6c63eb58390b79/ratis-common/src/main/java/org/apache/ratis/protocol/RaftPeer.java#L175

> How can we handle this now since the code has been merged?

HDDS-8009

Reply:
@adoroszlai Thanks! I'll create a patch shortly for [HDDS-8009](https://issues.apache.org/jira/browse/HDDS-8009).

Reply:
Thanks @adoroszlai for merging and uncovering this change.  With `getLeader() `no longer throwing an exception, we can cleanup the null check and handling of that condition.